dora/core/server/master/src/main/java/alluxio/master/job/LoadJobFactory.java

```diff
@@ -60,11 +62,23 @@ public Job<?> create() {
         .ofNullable(AuthenticatedClientUser.getOrNull())
         .map(User::getName);
 
+    Predicate<UfsStatus> predicate = Predicates.alwaysTrue();
+    if (options.hasFileFilterRegx()) {
+      String regxPatternStr = options.getFileFilterRegx();
+      if (regxPatternStr != null && !regxPatternStr.isEmpty()) {
+        alluxio.proto.journal.Job.FileFilter.Builder builder =
+            alluxio.proto.journal.Job.FileFilter.newBuilder()
+                .setName("fileNamePattern").setValue(regxPatternStr);
+        FilePredicate filePredicate = FilePredicate.create(builder.build());
+        predicate = filePredicate.getUfsStatusPredicate();
+      }
+    }
+
     UnderFileSystem ufs = mFs.getUfsManager().getOrAdd(new AlluxioURI(path),
         UnderFileSystemConfiguration.defaults(Configuration.global()));
     Iterable<UfsStatus> iterable = new UfsStatusIterable(ufs, path,
         Optional.ofNullable(AuthenticatedClientUser.getOrNull()).map(User::getName),
-        Predicates.alwaysTrue());
+        predicate);
     return new DoraLoadJob(path, user, UUID.randomUUID().toString(), bandwidth, partialListing,
```

Comment:
We also want to put this filter information into DoraLoadJob so we can turn this job into a journal entry.

Reply:
Updated.